mayastor/src/bdev/nvmx/controller.rs

```diff
@@ -927,13 +929,34 @@ pub(crate) async fn destroy_device(name: String) -> Result<(), NexusBdevError> {
 
     // Notify the listeners.
     debug!(?name, "notifying listeners about device removal");
-    let controller = carc.lock();
-    let num_listeners = controller.notify_event(DeviceEventType::DeviceRemoved);
-    debug!(
-        ?name,
-        ?num_listeners,
-        "listeners notified about device removal"
-    );
+    {
+        let controller = carc.lock();
+        let num_listeners =
+            controller.notify_event(DeviceEventType::DeviceRemoved);
+        debug!(
+            ?name,
+            ?num_listeners,
+            "listeners notified about device removal"
+        );
+    }
+
+    let mut carc = carc;
+    loop {
+        match Arc::try_unwrap(carc) {
+            Ok(i) => {
+                drop(i);
+                break;
+            }
+            Err(ret) => {
+                warn!(?name, "delaying controller destroy");
+                let rx = mayastor_sleep(std::time::Duration::from_secs(1));
```

Comment:
From the logs the channels are usually ready very quickly when we hit this, maybe start with a smaller number? 100ms or something like that?

Reply:
On a 100 iteration test, 100ms does seem to be more than sufficient as there were 11 instances of controller destruction being delayed and all on separate iterations. However, this is on an otherwise idle system so I've settled on 250ms as hopefully a long enough delay, without being too long.